scaleway/resource_compute_instance_volume.go

```diff
@@ -0,0 +1,203 @@
+package scaleway
+
+import (
+	"fmt"
+
+	"github.com/dustin/go-humanize"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	namesgenerator "github.com/scaleway/scaleway-sdk-go/namegenerator"
+)
+
+func resourceScalewayComputeInstanceVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayComputeInstanceVolumeCreate,
+		Read:   resourceScalewayComputeInstanceVolumeRead,
+		Update: resourceScalewayComputeInstanceVolumeUpdate,
+		Delete: resourceScalewayComputeInstanceVolumeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "the name of the volume",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if new == "" {
+						return true
+					}
+					return old == new
+				},
+			},
+			"size": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "the size of the volume in humab readable format (e.g. 20GB)", // TODO: human readable
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					_, err := humanize.ParseBytes(val.(string))
+					if err != nil {
+						errs = append(errs, fmt.Errorf("couldn't parse volume size: %s", err))
+					}
+					return
+				},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if old == new {
+						return true
+					}
+
+					oldInBytes, err := humanize.ParseBytes(old)
+					if err != nil {
+						return false
+					}
+					newInBytes, err := humanize.ParseBytes(new)
+					if err != nil {
+						return false
+					}
+
+					return newInBytes == oldInBytes
+				},
+			},
+			// TODO handle snapshot, base_volume
+			"server_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "the server associated with this volume",
```

Comment:
add `ValidateFunc: validationUUID(),`

Reply:
It's a computed field, so I cannot ad a validate function.